plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java

```diff
@@ -118,6 +123,22 @@ protected IStatus doRun(IProgressMonitor monitor) {
 		return Status.OK_STATUS;
 	}
 
+	private boolean isUpdate(Connection connection, String project, String name) {
+		try {
+			IDeploymentConfig dc = connection.getResource(ResourceKind.DEPLOYMENT_CONFIG, project, name);
+			IDeploymentImageChangeTrigger trigger = (IDeploymentImageChangeTrigger) dc.getTriggers().stream()
+					.filter(t->DeploymentTriggerType.IMAGE_CHANGE.equals(t.getType())).findFirst().orElse(null);
+			return trigger != null && 
+				ResourceKind.IMAGE_STREAM_TAG.equals(trigger.getKind()) && 
+				connection.getResource(ResourceKind.IMAGE_STREAM, trigger.getNamespace(), trigger.getFrom().getName()) != null;
+		}catch(OpenShiftException e) {
+			if(e.getStatus() != null && e.getStatus().getCode() != IHttpConstants.STATUS_NOT_FOUND) {
```

Comment:
if status is null, it will swallow the exception. is it intended? Else it should be e.getStatus() == null || e.getStatus().getCode() != IHttpConstants.STATUS_NOT_FOUND